src/components/EditInputGroup.vue

```diff
@@ -19,42 +19,115 @@
 
 <script>
 export default {
-    name: "EditInputGroup",
-    props: ['fieldType','fieldName','fieldLabel'],
+    name: "EditInputGroup",  
+    props: {
+        fieldType: String,
+        fieldName: String,
+        fieldLabel: String,
+        // If true, add new fields, for newlines in pasted data
+        createFieldsOnNewlines: {
+            type: Boolean,
+            default: false
+        },
+    },
     data () {
         return {
+            // helper variables
+            contentPasted: false
         }
     },
     methods: {
-        addNew: function() {
-            // This is a dirty hack, but Vue components update with a slight delay so you
-            // can't just straight up go and set focus here
-            let nextFocus = this.$parent.recipe[this.fieldName].length
-            this.$parent.addEntry(this.fieldName)
-            let failSafe = 2500
-            let $ul = $(this.$refs['list'])
-            let $this = this
-            let focusMonitor = window.setInterval(function() {
-                if ($ul.children('li').length > nextFocus) {
-                    if ($this.fieldType === 'text') {
-                        $ul.children('li').eq(nextFocus).find('input').focus()
-                    } else if ($this.fieldType === 'textarea') {
-                        $ul.children('li').eq(nextFocus).find('textarea').focus()
+        /* if index = -1, element is added at the end
+         * if focusAfterInsert=true, the element is focussed after inserting
+         * the content is inserted into the newly created field
+         **/
+        addNew: function(index = -1, focusAfterInsert = true, content = '') {
+            if (index === -1) {
+                index = this.$parent.recipe[this.fieldName].length
+            }
+            this.$parent.addEntry(this.fieldName, index, content)
+
+            if (focusAfterInsert) {
+                let $ul = $(this.$refs['list'])
+                let $this = this
+                this.$nextTick(function() {
+                    if ($ul.children('li').length > index) {
+                        if ($this.fieldType === 'text') {
+                            $ul.children('li').eq(index).find('input').focus()
+                        } else if ($this.fieldType === 'textarea') {
+                            $ul.children('li').eq(index).find('textarea').focus()
+                        }
                     }
-                    window.clearInterval(focusMonitor)
-                }
-                failSafe -= 100
-                if (!failSafe) {
-                    window.clearInterval(focusMonitor)
-                }
-            }, 100)
+                })
```

Comment:
@sam-19 btw, I'm not sure if the "dirty hack" part for focussing the new field was yours, but I just wanted to point this section out for deeper review and maybe future reference. for me this seems to work fine! :)

Reply:
That was probably just grabbed from some old project when I wasn't yet familiar with the tick-based update method Vue uses. $nextTick is definitely the way to go here.